lib/active_record/sharding/abstract_repository.rb

```diff
@@ -0,0 +1,30 @@
+module ActiveRecord
+  module Sharding
+    class AbstractRepository
+
+      private
+
+        def generate_model_for_shard(connection_name)
+          base_class_name = @base_class.name
+          class_name = generate_class_name connection_name
+
+          model = Class.new(base_class) do
+            self.table_name = base_class.table_name
+
+            module_eval <<-RUBY, __FILE__, __LINE__ + 1
+                  def self.name
+                    "#{base_class_name}::#{class_name}"
+                  end
+                RUBY
+          end
+
+          model.class_eval { establish_connection(connection_name) }
+          model
+        end
+
+        def generate_class_name(connection_name)
```

Comment:
Lint/UnusedMethodArgument: Unused method argument - `connection_name`. If it's necessary, use `_` or `_connection_name` as an argument name to indicate that it won't be used. You can also write as `generate_class_name(*)` if you want the method to accept any arguments but don't care about them.